huaweicloud/resource_huaweicloud_dms_rabbitmq_instance_test.go

```diff
@@ -0,0 +1,267 @@
+	var instance instances.Instance
+	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	resourceName := "huaweicloud_dms_rabbitmq_instance.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheckEpsID(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDmsRabbitmqInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDmsRabbitmqInstance_withEpsId(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDmsRabbitmqInstanceExists(resourceName, instance),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "engine", "rabbitmq"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key", "value"),
+					resource.TestCheckResourceAttr(resourceName, "tags.owner", "terraform"),
+					resource.TestCheckResourceAttr(resourceName, "enterprise_project_id", HW_ENTERPRISE_PROJECT_ID_TEST),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckDmsRabbitmqInstanceDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*config.Config)
+	dmsClient, err := config.DmsV2Client(HW_REGION_NAME)
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud dms instance client: %s", err)
+	}
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "huaweicloud_dms_rabbitmq_instance" {
+			continue
+		}
+
+		_, err := instances.Get(dmsClient, rs.Primary.ID).Extract()
+		if err == nil {
+			return fmt.Errorf("The Dms rabbitmq instance still exists.")
+		}
+	}
+	return nil
+}
+
+func testAccCheckDmsRabbitmqInstanceExists(n string, instance instances.Instance) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		config := testAccProvider.Meta().(*config.Config)
+		dmsClient, err := config.DmsV2Client(HW_REGION_NAME)
+		if err != nil {
+			return fmt.Errorf("Error creating HuaweiCloud dms instance client: %s", err)
+		}
+
+		v, err := instances.Get(dmsClient, rs.Primary.ID).Extract()
+		if err != nil {
+			return fmt.Errorf("Error getting HuaweiCloud dms rabbitmq instance: %s, err: %s", rs.Primary.ID, err)
+		}
+
+		if v.InstanceID != rs.Primary.ID {
+			return fmt.Errorf("The Dms rabbitmq instance not found.")
+		}
+		instance = *v
+		return nil
+	}
+}
+
+func testAccDmsRabbitmqInstance_basic(rName string) string {
+	return fmt.Sprintf(`
+data "huaweicloud_dms_az" "test" {}
+
+data "huaweicloud_vpc" "test" {
+  name = "vpc-default"
+}
+
+data "huaweicloud_vpc_subnet" "test" {
+  name = "subnet-default"
+}
+
+data "huaweicloud_dms_product" "test" {
+  engine        = "rabbitmq"
+  instance_type = "cluster"
+  version       = "3.7.17"
+}
+
+resource "huaweicloud_networking_secgroup" "test" {
+  name        = "%s"
+  description = "secgroup for rabbitmq"
+}
+
```

Comment:
suggest to wrap L135-L154 into a base config function.

Reply:
fixed